google/cloud/bigquery/_pandas_helpers.py

```diff
@@ -444,7 +444,14 @@ def augment_schema(dataframe, current_bq_schema):
             continue
 
         arrow_table = pyarrow.array(dataframe[field.name])
-        detected_type = ARROW_SCALAR_IDS_TO_BQ.get(arrow_table.type.id)
+
+        if arrow_table.type.id == 25:
```

Comment:
Is there a constant we can use to derive this number? I'm a bit wary of having the magic number 25 in tests.

Reply:
Or better yet, can we use [pyarrow.types.is_list(arrow_table.type)](https://arrow.apache.org/docs/python/generated/pyarrow.types.is_list.html#pyarrow.types.is_list)?

Reply:
Will implement this tomorrow.

Reply:
Done.